### PR TITLE
EOS-17361 : Porting of BG consumer to read configurations from conf store.

### PR DESCRIPTION
**Pull request by pranavPawar07**
Signed-off-by: pranav <pranav.pawar@seagate.com>

### CODE REVIEW

File: s3backgrounddelete/s3backgrounddelete/cortx_s3_config.py [hunk 1]
```diff
@@ -95,106 +96,188 @@ def cache_credentials(self):
...
-        if 'logconfig' in self._config and self._config['logconfig']['logger_directory']:
-            return self._config['logconfig']['logger_directory']
-        else:
+        #if 'logconfig' in self._config and self._config['logconfig']['logger_directory']:
+        #    return self._config['logconfig']['logger_directory']
+        #else:
+        #    raise KeyError(
+        #        "Could not parse logger directory path from config file " +
+        #        self._conf_file)
+        try:
```
Comment (nileshgovande) [inline s3backgrounddelete/s3backgrounddelete/cortx_s3_config.py:117]: if and try do not go together - use only one

File: s3backgrounddelete/s3backgrounddelete/cortx_s3_config.py [hunk 2]
```diff
@@ -74,7 +74,8 @@ def _load_and_fetch_config(self):
                 " it doesn't have read access")
             sys.exit()
         with open(self._conf_file, 'r') as file_config:
```
Comment (nileshgovande) [inline s3backgrounddelete/s3backgrounddelete/cortx_s3_config.py:76]: do not 'open' the file - kick off - confstore class

Comment (nileshgovande) [reply s3backgrounddelete/s3backgrounddelete/cortx_s3_config.py:76]: check example code of confstore

File: s3backgrounddelete/s3backgrounddelete/cortx_s3_config.py (hunk 1 above)
Comment (pranavPawar07) [reply s3backgrounddelete/s3backgrounddelete/cortx_s3_config.py:122]: done

File: s3backgrounddelete/s3backgrounddelete/cortx_s3_config.py (hunk 2 above)
Comment (pranavPawar07) [reply s3backgrounddelete/s3backgrounddelete/cortx_s3_config.py]: done

**Comment by gowthamchinna** [comment]
run mini-provisioning

File: s3backgrounddelete/ut/cortx_s3_config_tests.py [hunk 3]
```diff
@@ -24,9 +24,15 @@
...
 from s3backgrounddelete.cortx_s3_config import CORTXS3Config
+from s3confstore.cortx_s3_confstore import S3CortxConfStore
 
 CONFIG_LOG_DIR = "/var/log/seagate/s3/s3backgrounddelete"
+conf_home_dir = os.path.join(
+            '/', 'opt', 'seagate', 'cortx', 's3', 's3backgrounddelete')
+_conf_file = os.path.join(conf_home_dir, 'config.yaml')
+conf_url = 'yaml://' + _conf_file
 
+s3confstore = S3CortxConfStore(conf_url, "dummy")
```
Comment (AtitaShirwaikar) [inline s3backgrounddelete/ut/cortx_s3_config_tests.py:35]: Constructor of CORTXS3Config() will call this.
Is there a need to call S3CortxConfStore externally?

Comment (pranavPawar07) [reply s3backgrounddelete/ut/cortx_s3_config_tests.py]: fixed. Please check. @AtitaShirwaikar

**Comment by pranavPawar07** [comment]
S3Server-pre-merge-test (**PASSED**) : http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Component-Test/job/S3Server-PreMerge-Test/162/

**Comment by nileshgovande** [comment]
http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Component-Test/job/S3Server-PreMerge-Test/162/ - passed

**Pull request merged**